orderer/common/multichannel/registrar_test.go

```diff
@@ -1543,6 +1571,119 @@ func TestRegistrar_RemoveChannel(t *testing.T) {
 		require.Nil(t, registrar.GetChain("channel-im-not-a-member-of"))
 		require.NotContains(t, ledgerFactory.ChannelIDs(), "channel-im-not-a-member-of")
 	})
+
+	t.Run("remove system channel ledger fails", func(t *testing.T) {
+		setup(t)
+		defer cleanup()
+
+		newLedger(ledgerFactory, "raft-sys-channel", genesisBlockSysRaft)
+		registrar := NewRegistrar(config, ledgerFactory, mockCrypto(), &disabled.Provider{}, cryptoProvider, dialer)
+		registrar.Initialize(mockConsenters)
+		require.NotNil(t, registrar.GetChain("raft-sys-channel"))
+		require.Contains(t, ledgerFactory.ChannelIDs(), "raft-sys-channel")
+
+		os.RemoveAll(filepath.Join(tmpdir, "pendingops", "remove"))
+
+		err := registrar.RemoveChannel("raft-sys-channel")
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "error removing ledger for system channel raft-sys-channel")
+
+		// After removing the system channel, it no longer exists in the registrar or the ledger
+		require.NotNil(t, registrar.SystemChannelID())
+		require.Contains(t, ledgerFactory.ChannelIDs(), "raft-sys-channel")
+
+	})
+
+	t.Run("remove non-systemchannel members fails", func(t *testing.T) {
+		setup(t)
+		defer cleanup()
+
+		newLedger(ledgerFactory, "raft-sys-channel", genesisBlockSysRaft)
+		registrar := NewRegistrar(config, ledgerFactory, mockCrypto(), &disabled.Provider{}, cryptoProvider, dialer)
+		registrar.Initialize(mockConsenters)
+		require.NotNil(t, registrar.GetChain("raft-sys-channel"))
+		require.Contains(t, ledgerFactory.ChannelIDs(), "raft-sys-channel")
+
+		genesisBlockAppRaftNonMember := appBootstrapper.GenesisBlockForChannel("channel-im-not-a-member-of")
+		require.NotNil(t, genesisBlockAppRaftNonMember)
+		createLedgerAndChain(t, registrar, ledgerFactory, genesisBlockAppRaftNonMember, "channel-im-not-a-member-of")
+
+		require.NotNil(t, registrar.GetChain("channel-im-not-a-member-of"))
+		require.Contains(t, ledgerFactory.ChannelIDs(), "channel-im-not-a-member-of")
+		consenter.IsChannelMemberStub = func(b *cb.Block) (bool, error) {
+			os.RemoveAll(filepath.Join(tmpdir, "pendingops", "remove"))
+			if bytes.Compare(b.Header.DataHash, genesisBlockAppRaft.Header.DataHash) == 0 {
+				return true, nil
+			}
+			return false, nil
+		}
+
+		err := registrar.RemoveChannel("raft-sys-channel")
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "failed removing ledger for channel channel-im-not-a-member-of")
+
+		// After removing the system channel, it no longer exists in the registrar or the ledger
+		require.Empty(t, registrar.SystemChannelID())
+		require.Nil(t, registrar.SystemChannel())
+		require.Nil(t, registrar.GetChain("raft-sys-channel"))
+		require.NotContains(t, ledgerFactory.ChannelIDs(), "raft-sys-channel")
+
+		// channels the orderer wasn't a member no longer exist
```

Comment:
again, probably cut & paste error, please check all the comments here...

Reply:
🧹  removed.